### ISSUE TITLE
[Task]: Show the hovered option to bind preview in the input 

### ISSUE THREAD

**Issue by rishabhrathod01** [appsmithorg/appsmith#31112]
### SubTasks

![image](https://github.com/appsmithorg/appsmith/assets/23132741/e79aaef6-5085-4312-83f9-683cec465cd7)

**Comment by AmanAgarwal041** [comment]
Previously, we hoped to offer a preview of the binding in the codeeditor input when hovering over any proposed query hints, but it wasn't possible for a variety of reasons.
- Input changes cause linting, evaluations, popovers, and the slash command menu to close, among other things.
- Several redux states are modified.
As a result, this could be complex and negatively impact performance. As a result, we devised a variety of additional ways to improve learnability, including: 
- Displaying tooltips (had some tooltip persisting issues)
- Showing the binding preview instead of the proposed query names on hover (which influenced the ux)
- Displaying the binding preview as a description for each suggested query term (which may consume a lot of space).
To summarise the preceding observations and discussion, we agreed to put this task on hold for a bit until we had a more satisfactory solution.
cc @rohan-arthur @carinanfonseca
